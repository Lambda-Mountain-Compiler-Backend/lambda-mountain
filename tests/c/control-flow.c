
int main() {
   int x, y;
   if(x){ y; };
   if(x){ y; } else { y; };
   for(;;){ x; };
   for(int z;;){ x; };
   for(;x;){ x; };
   for(;;x){ x; };
   for(int z; x;){ x; };
   for(int z;;x){ x; };
   for(;x;x){ x; };
   for(int z;x;x){ x; };
   while(x){ x; };
   do{x;} while(x);
   switch(1){
      case 1: continue;
      default: break;
   };
   return;
   return 0;
   label_0:
   goto label_0;
}
