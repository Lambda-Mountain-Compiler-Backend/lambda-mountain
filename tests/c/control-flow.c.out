
int main(){
   int x ;
   int y ;
   int z ;
   int z2 ;
   int z3 ;
   int z4 ;
   int z5 ;
   if(x){{y;}}else{};
   if(x){{y;}}else{{y;}};
   for(;;){{(x+1);}};
   for(;;){{(x+2);}};
   for(({z2=2;({});});;){{(x+3);}};
   for(;x;){{(x+4);}};
   for(;;x){{(x+5);}};
   for(;x;){{(x+6);}};
   for(;;x){{(x+7);}};
   for(;x;x){{(x+8);}};
   for(;x;x){{(x+9);}};
   while(x){{x;}};
   do{{x;}}while(x);
   switch(1){{case 1:continue;default:break;}};
   return;
   return 0;
   label_0:;
   goto label_0;
}
